Multi lang support

BHess2653 opened the pull request:
Currently working but is not using the database. This is not ready for release at the moment but is ready for review any feedback to help guide this feature toward the next release will help.

- Migrations & Seeders(commented out in first_run.sh) are made
- process.env.LANG added to gulpfile

Commit:
started a migration for messages

Commit:
removed create-messages

Commit:
created migration, model, and seeder

Commit:
added localize for language support

Commit:
added localize & language from botUtils

Commit:
added localize & language from botUtils

Commit:
added localize & language from botUtils

Commit:
added language to botUtils

Commit:
added localize & language from botUtils

Commit:
removed changes to file from last commit

Commit:
added LANG=en

Commit:
changed language() to process.env.LANG

Commit:
fixed errors removed unused code

Commit:
updated en - EN etc

Commit:
changed en - EN etc

Commit:
created & updated files in db folder

mattpezzente commented on bot/client.js:
```diff
@@ -5,9 +5,9 @@ const Localize = require('localize');
+    AL: 'v1.4.0 lakwnd awkjjbz:\n\n\t** <> - yfigijgkjg jhckjfiuf\t\t[] - hvkjfnkfcjh yfifkjvk**',
   },
 });
 myLocale.setLocale(process.env.LANG);
```
Set a fallback for this in case a .env does not have a LANG set by default.

mattpezzente reviewed (changes requested):
Overall looks good, I could not think of another way to implement this, as I know doing live translations would be a large undertaking in itself. I know that this is still in progress, but definitely make sure the user is able to set the language if possible.

Lastly, just make sure to add comments for readability.

Commit:
updated readme sample env to have LANG=en

Commit:
removed localize

Commit:
removed english, spanish, and alien files

Commit:
created messages.json for message language types

Commit:
removed localize and added messages.json from locale folder

BHess2653 commented:
This is now ready for release 2 and needs more code reviews.

- messages in en, es, and al are now stored in the bot/locale/messages.json
- added messages.json to controllers
- fixed typo in !addRoles help message from Rolea to Roles

brennondenny commented on readme.md:
````diff
@@ -23,6 +23,7 @@ Create an ```.env``` file with the following sensitive information. Replace the
 ```
 NODE_ENV=development
 DEBUG_MODE=3
+LANG=en
````
Consider Changing all instances on LANG to LANGUAGE so it will be easily identified in the codebase and the .ENV file

BHess2653 replied:
ok changes have been made.

brennondenny reviewed (changes requested):
Functionality works with the different commands in message.json, make sure to add LANGUAGE to the README's .ENV setup guide.

Commit:
changed LANG to LANGUAGE

Commit:
updated readme on how to set up LANGUAGE

Commit:
added missing messages from other controllers

Commit:
finished adding messages.json to the rest of the controllers

BHess2653 commented:
Updated messages.json with all messages from controllers.
Ready for Release 2

Commit:
fixed message error

brennondenny committed:
Merge branch 'Release2' into multiLang_Hess

reactivepixel commented:
There's a lot of redundancy in how you're calling and passing in the environmental variables. It's an environment variable... It should not need passed into anything. 

Also, adding a further level of organization to your .json file that allows the messages be organized according by controller as well will bring that I line with my expectations. 

Those items said... Good implementation!
